code/game/objects/machinery/deployable.dm

```diff
@@ -1,15 +1,109 @@
 /obj/machinery/deployable
-	name = "deployable"
-	desc = "deployable"
-	icon = 'icons/obj/objects.dmi'
+	///Item that is deployed to create src
+	var/obj/item/internal_item
+	///Flags for machine functions
+	var/deploy_flags = NONE
```

Comment:
This var should die, because it is on the component as well.

This means you have to move the rotate code in the component, as well with the disassemble code.

Which both seems to belong more to the component anyway